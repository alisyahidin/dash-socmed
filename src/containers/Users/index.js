import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Container, Row, Col } from 'react-bootstrap'

import withNavbar from '../../hoc/withNavbar'
import Card from '../../components/Card/'
import User from '../../components/User/'

import { fetchUser } from '../../actions/user'

class Users extends Component {
  componentDidMount() {
    this.props.fetchUser()
  }

  render() {
    const { user } = this.props

    return (
      <Container>
        <Row>
          <Col md={{ span: 10, offset: 1 }}>
            <Card align="left">
              {user.data.map(user => (
                <User key={user.id} id={user.id} {...user} />
              ))}
            </Card>
          </Col>
        </Row>
      </Container>
    )
  }
}

const mapStateToProps = state => ({ ...state })

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    fetchUser
  }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(withNavbar(Users))