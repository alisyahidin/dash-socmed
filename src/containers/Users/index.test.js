import React from 'react'
import { shallow } from 'enzyme'
import Users from './index'

it('render <Users /> correctly', () => {
  const component = shallow(<Users />)

  expect(component).toMatchSnapshot()
})