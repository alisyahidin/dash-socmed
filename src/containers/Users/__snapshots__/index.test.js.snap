// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Users /> correctly 1`] = `
<Fragment>
  <withRouter(NavigationBar) />
  <div
    className="main-content"
  >
    <Users />
  </div>
</Fragment>
`;
